testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,263 @@
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean invokeAndCheck(Method method, Object[] params) {
+        try {
+            method.invoke(null, params);
+        } catch (InvocationTargetException ex) {
+            boolean isValid = validateException(method.getName(), ex);
+            return isValid;
+        } catch (IllegalAccessException e) {
+            throw Exceptions.wrappedCause(e);
+        }
+        return false;
+    }
+
+    private boolean validateException(String methodName, InvocationTargetException ex) {
+        final Throwable cause = ex.getCause();
+        checkException(cause);
+        final boolean result = isCorrectStackTraceElements(methodName, cause);
+        return result;
+    }
+
+    private void checkException(Throwable cause) {
+        final boolean isCorrectException = cause instanceof NullPointerException;
+        if (!isCorrectException) {
+            throw Exceptions.wrappedCause(cause);
+        }
+    }
+
+    private boolean isCorrectStackTraceElements(String methodName, Throwable cause) {
+        final StackTraceElement[] stackTraceElements = cause.getStackTrace();
+        final StackTraceElement preconditionsElement = stackTraceElements[0];
+        final boolean isPreconditionsClass = Preconditions.class.getName()
+                                                                .equals(preconditionsElement.getClassName());
+        if (!isPreconditionsClass) {
+            return false;
+        }
+
+        final StackTraceElement expectedUtilClassElement = stackTraceElements[1];
+        final boolean isCorrectMethod = isCorrectMethodName(methodName, expectedUtilClassElement);
+        if (!isCorrectMethod) {
+            return false;
+        }
+
+        final boolean isUtilClass = utilClass.getName()
+                                             .equals(expectedUtilClassElement.getClassName());
+        return isUtilClass;
+    }
+
+    private boolean isCorrectMethodName(String methodName, StackTraceElement expectedUtilClassElement) {
+        final boolean isCorrectMethodName = methodName.equals(expectedUtilClassElement.getMethodName());
+        return isCorrectMethodName;
+    }
+
+    /**
+     * Creates a new builder for the {@code NullToleranceTest}.
+     *
+     * @return the {@code Builder}
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    private void initPrimitiveValuesSet() {
+        primitiveValues.add(boolean.class);
+        primitiveValues.add(byte.class);
+        primitiveValues.add(short.class);
+        primitiveValues.add(int.class);
+        primitiveValues.add(long.class);
+        primitiveValues.add(char.class);
+        primitiveValues.add(float.class);
+        primitiveValues.add(double.class);
+    }
+
+    /**
+     * A builder for producing the {@link NullToleranceTest} instance.
+     */
+    public static class Builder {
```

Comment:
Please add getters and have a `NullToleranceTestBuilderShould` test to test the builder itself.
Check out some of the `Builder`s in Spine and how they are tested.